irmaseal-core/src/stream/tests.rs

```diff
@@ -29,47 +29,44 @@ impl Default for DefaultProps {
     }
 }
 
-fn seal(props: &DefaultProps, content: &[u8]) -> BigBuf {
+async fn seal<'a>(props: &DefaultProps, content: &[u8]) -> BigBuf {
     let mut rng = rand::thread_rng();
     let DefaultProps { i, pk, sk: _ } = props;
 
-    let mut buf = BigBuf::new();
-    {
-        let mut s = Sealer::new(i.clone(), &PublicKey(pk.clone()), &mut rng, &mut buf).unwrap();
-        s.write(&content).unwrap();
-    } // Force Drop of s.
+    let mut buf_writer = IntoAsyncWrite::from(BigBuf::new());
 
-    buf
+    let mut s = Sealer::new(i.clone(), &PublicKey(pk.clone()), &mut rng, &mut buf_writer)
+        .await
+        .unwrap();
+    s.seal(content).await.unwrap();
+
+    buf_writer.into_inner()
 }
 
-fn unseal(props: &DefaultProps, buf: &[u8]) -> (BigBuf, bool) {
+async fn unseal(props: &DefaultProps, buf: &[u8]) -> (BigBuf, bool) {
     let mut rng = rand::thread_rng();
     let DefaultProps { i, pk, sk } = props;
 
-    let bufr = SliceReader::new(&buf);
-    let (m, o) = OpenerSealed::new(bufr).unwrap();
+    let (m, o) = OpenerSealed::new(buf).await.unwrap();
     let i2 = &m.identity;
 
     assert_eq!(&i, &i2);
 
     let usk = ibe::kiltz_vahlis_one::extract_usk(&pk, &sk, &i2.derive().unwrap(), &mut rng);
+    let mut dst = IntoAsyncWrite::from(BigBuf::new());
```

Comment:
Somehow I have introduced an out of stack error in the tests. When I increase the `RUST_MIN_STACK` to `3000000` (I think 3MB) the problem is solved, so probably I use a large buffer somewhere. I think it is related to this `BigBuf` allocation. I got a bit stuck on this issue.